openeo/local/connection.py

```diff
@@ -270,6 +272,7 @@ def execute(
         *,
         validate: Optional[bool] = None,
         auto_decode: bool = True,
+        return_provenance: bool = False,
```

Comment:
You are adding a custom argument here to a more public API of `Connection.execute()`, which is fine as long as you call this `LocalConnection.execute()` yourself. But in general this method will be called automatically without this argument (because it's not in the official API), e.g.:

```python
cube = local_connection.load_collection(...)
res = cube.execute()  
```
the latter `execute()` is a method defined on `DataCube` and does not support `return_provenance`, let alone it will pass it properly to `LocalConnection.execute`